kuzzle-esp32/01-rgb-led/rgb-led.md

```diff
@@ -0,0 +1,333 @@
+    ledc_set_fade_with_time(LEDC_HIGH_SPEED_MODE, BLUE_PWM_CHANNEL, b_duty, LEDC_TRANSITION_TIME);
+
+    ledc_fade_start(LEDC_HIGH_SPEED_MODE, RED_PWM_CHANNEL, LEDC_FADE_NO_WAIT);
+    ledc_fade_start(LEDC_HIGH_SPEED_MODE, GREEN_PWM_CHANNEL, LEDC_FADE_NO_WAIT);
+    ledc_fade_start(LEDC_HIGH_SPEED_MODE, BLUE_PWM_CHANNEL, LEDC_FADE_NO_WAIT);
+
+    _publish_light_state(); // Publish new complete state
+}
+```
+
+## Get the Code
+
+You can get the whole source code for the rgb light from my github repository: <https://github.com/etrousset/instructable.git>. It's located on the 'rbg-light' branch. You can get it using the following git command:
+
+``` console
+$ git clone -b rgb-light https://github.com/etrousset/instructable.git
+```
+
+## Visualize Device State
+
+![visualise device state](img/admin-console-device-state.png)
+
+For this last step we will use Kuzzle Console to visualize and change the state of the RGB light.
+
+When booting, the device will publish its current state. You should be able to visualize it in Kuzzle Console as shown in the picture.
+
+To do so, open Kuzzle Console and connect it to your Kuzzle server instance. Then open the **iot** index, and then the **device-state** collection.
+
+If your device booted up, you should see a document representing the state of your RBG light. The last step is to control the RBG light through Kuzzle server. The easiest way to do this for now is using Kuzzle Console.
+
+## Get in Control...
+
+In this last step, we'll change the color of the RGB light by creating a new document in device-state collection.
+
+We will mark this document avec being a 'partial_state', meaning the device will have to merge it with its current state in order to get its new complete state. This allow changing only one parameter of the state with out having to publish the whole state. Also, the device is responsible for accepting or not the state change.
+
+In the case where it will update its state (turning ON or OFF, changing color) it will then publish its new complete state.
+
+Click on the **Create** button and enter the following Json:
+``` JSON
+{
+  "device_id": "30AEA480838D",
+  "partial_state": true,
+  "state": {
+    "r": 0
+  }
+}
+```
+
+The **device_id** field must be the same as the one from the document published in the previous step, it is built using the device MAC address.
+
+![new state](img/admin-console-new-state.png)
+
+Once you click on the Create button to validate your document, the RGB light should turn to bright blue. When you refresh your browser, you should now have 3 documents:
+
+* The initial one with r,g and b all set to 255,
+* The partial one you entered to turn r to 0
+* The new complete state published by the device with r set to 0 and g and b still set to 255
+
+![kuzzle](img/lanceur-backend-1900.jpg)
+
+## Going Further
+
+Check out the Kuzzle JS SDK to control your RGB light using an Javascript application: <http://docs.kuzzle.io/guide/getting-started/>
```

Comment:
`[...] using a Javascript application: [...]`

Also, the link should point directly to the SDK guide: https://docs.kuzzle.io/guide/getting-started/#fun-with-sdks